client/src/components/cogeneration-side-bar.tsx

```diff
@@ -0,0 +1,238 @@
+Permission to use, copy, modify, and distribute this software and its documentation for educational, research and non-profit purposes, without fee, and without a written agreement is hereby granted, provided that the above copyright notice and subject to the full license file found in the root of this software deliverable. Permission to make commercial use of this software may be obtained by contacting:  USC Stevens Center for Innovation University of Southern California 1150 S. Olive Street, Suite 2300, Los Angeles, CA 90115, USA Email: accounting@stevens.usc.edu
+
+The full terms of this copyright and license should always be found in the root directory of this software deliverable as "license.txt" and if these terms are not found with this software, please contact the USC Stevens Center for the full license.
+*/
+import React, { useContext } from "react";
+import CallResponseLog from "./call-response-log";
+import ViewPrompts from "./view-prompts";
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogTitle,
+  Grid,
+  IconButton,
+  List,
+  ListItem,
+} from "@mui/material";
+import {
+  Save as SaveIcon,
+  ArrowBack as ArrowBackIcon,
+  ArrowBackIosNew as BackIcon,
+  ArrowForwardIos as ForwardIcon,
+  ReceiptLong as ReceiptLongIcon,
+  RateReview as RateReviewIcon,
+} from "@mui/icons-material";
+import { DrawerHeader, Drawer } from "helpers/lessonsHelpers";
+
+import { ToastContainer, toast } from "react-toastify";
+import { navigate } from "gatsby";
+
+interface SideBarClasses {
+  drawer: string;
+  actionFooter: string;
+  button: string;
+}
+import CogenerationContext from "context/cogeneration";
+
+export function CogenerationSideBar(props: {
+  classes: SideBarClasses;
+}): JSX.Element {
+  const { classes } = props;
+  const [savePopUp, setSavePopUp] = React.useState(false);
+  const context = useContext(CogenerationContext);
+  if (!context) {
+    throw new Error("SomeComponent must be used within a CogenerationProvider");
+  }
+
+  const [drawerOpen, setDrawerOpen] = React.useState(true);
+
+  const handleDrawerChange = () => {
+    setDrawerOpen(!drawerOpen);
+  };
+
+  function handleDiscard() {
+    navigate(`/lessons`);
+  }
+
+  function handleSavePopUp(open: boolean): void {
+    setSavePopUp(open);
+  }
+
+  function handleSaveExit(): void {
+    saveChanges();
+    navigate(`/lessons`);
+  }
+
+  function handleSaveContinue(): void {
+    saveChanges();
+    handleSavePopUp(false);
+  }
+
+  function saveChanges(): void {
+    toast("Saving...");
+    toast("Success!");
+  }
+
+  const [openLog, setOpenLog] = React.useState(false);
+  const [selectedValue, setSelectedValue] = React.useState("");
+
+  const [openPrompts, setOpenPrompts] = React.useState(false);
+  const [selectedPrompt, setSelectedPrompt] = React.useState("");
+  const handleClickOpenLog = () => {
+    setOpenLog(true);
+  };
+  const handleClickOpenPrompts = () => {
+    setOpenPrompts(true);
+  };
+
+  const handleClosePrompts = () => {
```

Comment:
looks like this is using a hacky way to work around reacts async state updates. You could instead just merge the state values (openPrompts and selectedPrompt) into a single state value (as an object) to be able to update both at the same time.